lib/logstash/outputs/s3.rb

```diff
@@ -147,6 +148,9 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
 
   # Set the directory where logstash will store the tmp files before sending it to S3
   # default to the current OS temporary directory in linux /tmp/logstash
+  #
+  # NOTE: the reason we do not have a unique (isolated) temporary directory as a default, to support multiple plugin instances,
+  # is that we would have to rely on something static that does not change between restarts (e.g. a user set id => ...).
   config :temporary_directory, :validate => :string, :default => File.join(Dir.tmpdir, "logstash")
```

Comment:
was thinking of ways to warn users if there are multiple outputs running against the same dir (with `restore => true`).

the only thing I could think of would be to keep a `/tmp/logstash/my-tmp-dir/.logstash-output-s3` file around (deleted upon plugin stop) and than checking if there's another such file on start or in the parent dirs (e.g. `/tmp/logstash/.logstash-output-s3`) if so warning that restore won't work as expected ... but it feels a bit too much, thoughts?

Reply:
It's a tricky one because it's racy - we have no guarantee of the order of the plugin output creation, so the plugin attempting to write to the child folder could be created first - ie `/tmp/logstash/sub` can be created before `/tmp/logstash/`, so this test wouldn't flag a warning in this case.

Let's create a separate issue for this problem, and keep this one for the doc changes.